python/zarrs/pipeline.py

```diff
@@ -32,10 +32,40 @@
 )
 
 
-@dataclass(frozen=True)
+def get_codec_pipeline_impl(codec_metadata_json: str) -> CodecPipelineImpl:
+    return CodecPipelineImpl(
+        codec_metadata_json,
+        validate_checksums=config.get("codec_pipeline.validate_checksums", None),
+        # TODO: upstream zarr-python array.write_empty_chunks is not merged yet #2429
+        store_empty_chunks=config.get("array.write_empty_chunks", None),
+        chunk_concurrent_minimum=config.get(
+            "codec_pipeline.chunk_concurrent_minimum", None
+        ),
+        chunk_concurrent_maximum=config.get(
+            "codec_pipeline.chunk_concurrent_maximum", None
+        ),
+        num_threads=config.get("threading.max_workers", None),
+    )
```

Comment:
Not sure even what the promise is from zarr re: the stability of values fetched from the config so not sure if this is a potential source of bugs or not

Reply:
It is still up in the air: see https://github.com/zarr-developers/zarr-python/pull/2429#discussion_r1811364395

Reply:
I'll add a comment with a warning for the next person then in our docs about this fact and then maybe ask on the zulip but I think that shouldn't block this PR

Reply:
See: https://ossci.zulipchat.com/#narrow/channel/423692-Zarr/topic/Config.20Values.  Seems like it's best to leave this as-is for now

Reply:
Ok a good point raised about empty chunks: https://ossci.zulipchat.com/#narrow/channel/423692-Zarr/topic/Config.20Values/near/487023639

Probably should be pickled

Reply:
A good point raised: Given #**Zarr>Config Values@487049771**  

So will leave this as-is